app/models/mixins/retirement_mixin.rb

```diff
@@ -231,15 +231,34 @@ def valid_zone?
     respond_to?(:my_zone) && my_zone.present?
   end
 
-  def setup_event_hash(requester)
-    event_hash = {:retirement_initiator => "system"}
-    event_hash[retirement_base_model_name.underscore.to_sym] = self
-    event_hash[:host] = host if self.respond_to?(:host)
+  def system_context_requester
+    if try(:evm_owner_id)
+      User.find(evm_owner_id)
+    else
+      $log.info("System context defaulting to admin user because owner of #{name} not set.")
+      User.in_my_region.find_by(:userid => 'admin')
+    end
+  end
+
+  def q_user_info(q_options, requester)
     if requester
-      event_hash[:userid] = requester
-      event_hash[:retirement_initiator] = "user"
+      if requester.kind_of?(String)
+        requester = User.find_by(:userid => requester)
+      end
+      q_options[:user_id] = requester.id
+      q_options[:group_id] = requester.current_group.id if requester.current_group
```

Comment:
Heh - this makes me reference the code every time I see it - It is not necessary since the very next line is looking up the `current_group` and all, but I feel like this is better form. (sorry, think you probably introduced this from my suggestion before)

```suggestion
      q_options[:group_id] = requester.current_group_id if requester.current_group_id
```